python/dgl/dataloading/dataloader.py

```diff
@@ -336,8 +339,41 @@ def __len__(self):
         ) // self.batch_size
 
 
+def _numel_of_shape(shape):
+    return reduce(operator.mul, shape, 1)
+
+
+def _init_gpu_caches(graph, gpu_caches):
+    if not hasattr(graph, "_gpu_caches"):
+        graph._gpu_caches = {}, {}
```

Comment:
A tuple of dict is a little confusing, can we replace it to a dict of dict with meaningful keys? like {"node": {}, "edge": {}}?